web/src/settings_org.js

```diff
@@ -516,11 +520,22 @@ export function discard_property_element_changes(elem, for_realm_default_setting
             break;
         case "emojiset":
         case "user_list_style":
-        case "stream_privacy":
             // Because this widget has a radio button structure, it
             // needs custom reset code.
-            $elem.find(`input[value='${CSS.escape(property_value)}']`).prop("checked", true);
+            set_radio_button_value($elem, property_value);
             break;
+        case "stream_privacy": {
+            set_radio_button_value($elem, property_value);
+
+            // Hide stream privacy warning banner
+            const $stream_permissions_warning_banner = $(
+                "#stream_permission_settings .stream-permissions-warning-banner .stream_privacy_warning",
+            );
+            if ($stream_permissions_warning_banner.is(":visible")) {
+                $stream_permissions_warning_banner.empty();
```

Comment:
The element that should be emptied is `$("#stream_permission_settings .stream-permissions-warning-banner")`. And to check whether the banner is shown you can use `$("#stream_permission_settings .stream-permissions-warning-banner").is(":empty")`.